src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_plugin.cpp

```diff
@@ -680,6 +710,68 @@ namespace xdp {
     aieOffloaders[deviceId] = std::make_tuple(aieTraceOffloader, aieTraceLogger, deviceIntf);
   }
 
+  void AieTracePlugin::setFlushMetrics(uint64_t deviceId, void* handle)
+  {
+    auto drv = ZYNQ::shim::handleCheck(handle);
+    if (!drv)
+      return;
+    auto aieArray = drv->getAieArray();
+    if (!aieArray)
+      return;
+    auto Aie = aieArray->getDevInst();
+    auto aieDevice = std::make_shared<xaiefal::XAieDev>(Aie, false);
```

Comment:
Random comment. Don't use shared_ptr unless ownership is being shared.  Is this a case where std::make_unique should have been used?

Reply:
Thanks for the feedback, Soren. The pointer actually is shared with our database.

Reply:
Hmm, I don't see it.  I'm curious where the sharing occurs?

Reply:
Hey @pgschuey  creating aieFAL class again might cause issues. We should store the shared pointer we created when we initialized trace and store it in database and reuse it.

Reply:
Another thought. When counters and trace both are used, we ideally should use same pointer in both aie trace and profile plugins.

Reply:
Submitted fix today that improves on this issue. Instances of XAieDevInst and XAie_Device are included in the static database and used for AIE trace and/or profiling. Makes for simpler code, too.